packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1618,6 +1589,77 @@ export async function copySequentialDatasets(nodes: ZoweDatasetNode[]) {
+                title: localize("paste.dataSet.InPrg", "Copying File(s)"),
+            },
+            () => {
+                for (const child of children) {
+                    ZoweExplorerApiRegister.getMvsApi(node.getProfile()).copyDataSetMember(
+                        { dsn: lbl, member: child.getLabel().toString() },
+                        { dsn: dsname, member: child.getLabel().toString() },
+                        { replace: replace === "replace" }
+                    );
+                }
+                return Promise.resolve();
+            }
+        );
+    });
+}
+
+export type replaceDstype = "ps" | "po" | "da" | "mem";
+export type shouldReplace = "replace" | "cancel" | "notFound";
+export async function determineReplacement(nodeProfile: zowe.imperative.IProfileLoaded, name: string, type: replaceDstype): Promise<shouldReplace> {
+    const mvsApi = ZoweExplorerApiRegister.getMvsApi(nodeProfile);
+    const options = { responseTimeout: nodeProfile.profile?.responseTimeout };
+    const stringReplace = localize("copyDataSet.replace.option1", "Replace");
+    const stringCancel = localize("copyDataSet.replace.option2", "Cancel");
+    let q: string = null;
+    let replace = false;
+    if (type === "mem") {
+        const dsname = name.split("(")[0];
+        const member = name.split("(")[1].slice(0, -1);
+        const res = await mvsApi.allMembers(dsname, options);
+        if (res?.success && res.apiResponse?.items.some((m) => m.member === member.toUpperCase())) {
+            q = localize("copyDataSet.replace.mem.question", "The dataset member already exists.\nDo you want to replace it?");
+            replace = stringReplace === (await api.Gui.showMessage(q, { items: [stringReplace, stringCancel] }));
+        }
+    } else {
+        const res = await mvsApi.dataSet(name, options);
+        if (res?.success && res.apiResponse?.items.length > 0) {
+            // Assume DA
+            q = localize("copyDataSet.replace.da.question", "The direct (DA) dataset already exists.\nDo you want to replace it?");
+            if (type === "ps") {
+                q = localize("copyDataSet.replace.ps.question", "The physical sequential (PS) dataset already exists.\nDo you want to replace it?");
+            } else if (type === "po") {
+                q = localize(
+                    "copyDataSet.replace.po.question",
+                    "The partitioned (PO) dataset already exists.\nDo you want to merge them while replacing any existing members?"
+                );
+            }
+            replace = stringReplace === (await api.Gui.showMessage(q, { items: [stringReplace, stringCancel] }));
+        }
+    }
+    // Sonar cloud code-smell :'(
+    const returnValueIfNotReplacing = q === null ? "notFound" : "cancel";
+    return replace ? "replace" : returnValueIfNotReplacing;
+}
+
+export async function _copyProcessor(nodes: ZoweDatasetNode[], type: replaceDstype, action: Function) {
```

Comment:
Should we define the type signature for the function parameter `action`? Just in case someone tries to call this function elsewhere (as it is exported)